ens/main.py

```diff
@@ -259,6 +260,12 @@ def setup_name(
             return self._setup_reverse(name, address, transact=transact)
 
     def resolve(self, name: str, get: str = 'addr') -> Optional[Union[ChecksumAddress, str]]:
+        warnings.warn(
```

Comment:
I don't feel super strongly about it, but I noticed that we have 681 new deprecation/future warnings in our tests, so circling back to the conversation we had this morning, I think it may be worth adding the `fn_name` argument to the signature here and then issuing a warning if it's not present. As a user it would drive me nuts to get a whole bunch of warnings and not have an upgrade path. So something like: 
```
def resolve(self, name: str, get: str = 'addr', fn_name=None) -> Optional[Union[ChecksumAddress, str]]:
    if fn_name is None:
        warnings.warn(In v6, the resolve() method will be made internal..., to suppress this warning in the meantime, use the `fn_name` keyword argument. In v6 you'll be able to pass it positionally.)
    ...
```
But also if we're pretty sure that no one is using this method in practice, then I guess it's just a personal problem :)

Reply:
Haha, no I think that is a great idea for `v5` I'll push this up.

Reply:
I'm curious though now that I'm taking a look... since the ``resolve()`` method will be made internal and changed to ``_resolve()``, I don't think this is the ideal migration route. Instead, I think the appropriate path(s) would be to change any uses of ``ns.resolve()`` to either ``ns.address()`` for the forward resolution, or ``ns.name()`` for the reverse resolution. That was kind of the point of getting rid of this as a public API - to avoid confusion and abstract the resolutions to their more specific, appropriate methods.

Thoughts?